app/src/main/java/com/arygm/quickfix/ui/search/Announcement.kt

```diff
@@ -1,33 +1,308 @@
+                  moveContentHorizontal = 10.dp,
+                  heightField = 42.dp,
+                  widthField = 360.dp,
+                  modifier = Modifier.testTag("categoryInput"),
+                  showLabel = true,
+                  label = {
+                    Text(
+                        "Category *",
+                        style = MaterialTheme.typography.headlineSmall,
+                        color = colorScheme.onBackground,
+                        modifier = Modifier.padding(start = 3.dp).testTag("categoryText"))
+                  })
+            }
+
+            Spacer(modifier = Modifier.padding(6.dp))
+
+            Column(
+                modifier = Modifier.fillMaxWidth().padding(start = 8.dp),
+            ) {
+              QuickFixTextFieldCustom(
+                  value = description,
+                  onValueChange = {
+                    description = it
+                    descriptionIsEmpty = description.isEmpty()
+                  },
+                  placeHolderText = "Describe the quickFix",
+                  placeHolderColor = colorScheme.onSecondaryContainer,
+                  shape = RoundedCornerShape(12.dp),
+                  moveContentHorizontal = 10.dp,
+                  heightField = 95.dp,
+                  widthField = 360.dp,
+                  modifier = Modifier.testTag("descriptionInput"),
+                  showLabel = true,
+                  label = {
+                    Text(
+                        "Description *",
+                        style = MaterialTheme.typography.headlineSmall,
+                        color = colorScheme.onBackground,
+                        modifier = Modifier.padding(start = 3.dp).testTag("descriptionText"))
+                  },
+                  singleLine = false)
+            }
+
+            Spacer(modifier = Modifier.padding(6.dp))
+
+            Column(
+                modifier = Modifier.fillMaxWidth().padding(start = 8.dp),
+            ) {
+              QuickFixTextFieldCustom(
+                  value = location,
+                  onValueChange = { location = it },
+                  placeHolderText = "Enter the location",
+                  placeHolderColor = colorScheme.onSecondaryContainer,
+                  shape = RoundedCornerShape(12.dp),
+                  moveContentHorizontal = 10.dp,
+                  heightField = 42.dp,
+                  widthField = 360.dp,
+                  modifier = Modifier.testTag("locationInput"),
+                  showLabel = true,
+                  label = {
+                    Text(
+                        "Location *",
+                        style = MaterialTheme.typography.headlineSmall,
+                        color = colorScheme.onBackground,
+                        modifier = Modifier.padding(start = 3.dp).testTag("locationText"))
+                  })
+            }
+
+            Spacer(modifier = Modifier.padding(10.dp))
+
+            QuickFixButtonWithIcon(
```

Comment:
I've refactored the QuickFixButton in #129, you can now add a trailing and a leading icon so no need for a new composable

Reply:
Perfect i will use it in the next pr.